cpu/arm7_common/include/atomic_utils_arch.h

```diff
@@ -0,0 +1,76 @@
+/*
```

Comment:
These atomic_utils_arch look awfully similar. Quick look shows AVR8 only has a subset, and some have extra bitband.
Would this be easy to dedup a bit more?

Reply:
I think it might be useful to have the freedom to implement them completely independent from everyone else. Right now, I'm considering to implement them in inline assembly as LLVM has no convenience builtin functions that work on types not marked as `_Atomic`.

Reply:
ok. I just thought that those impls mapping to "__atomic_store_N()" could be unified, but explicity also has its upsides.